window-copy.c

```diff
@@ -3048,31 +3068,66 @@ window_copy_search_jump(struct window_mode_entry *wme, struct grid *gd,
 		return (window_copy_search_jump(wme, gd, sgd,
 		    direction ? 0 : gd->sx - 1,
 		    direction ? 0 : gd->hsize + gd->sy - 1, fy, cis, 0,
-		    direction, regex, foundlen));
+		    direction, regex));
 	}
 	return (0);
 }
 
+static void
+window_copy_move_before_search_mark(struct window_copy_mode_data *data,
+    u_int *fx, u_int *fy, int wrapflag)
+{
+	struct screen  *s = data->backing;
+	u_int		at, start;
+
+	if (window_copy_search_mark_at(data, *fx, *fy, &start) == 0 &&
+	    data->searchmark[start] != 0)
+		while (window_copy_search_mark_at(data, *fx, *fy, &at) == 0 &&
```

Comment:
Enclose the `while` in braces.

Reply:
Ah, so the rule is more than 1 line, rather than more than one statement? I got it, thanks.

Reply:
I can't speak for the code style preferred by @topcat001 or @nicm but I always require braces around code blocks that follow `if`, `else`, `while`, `for`, etc. In my four decades of programming I have personally found two bugs due to missing braces. More importantly, I find the inconsistency makes code harder to read. So even this simple block requires braces, IMHO:

```
if (true) {
    do_something();
}
```

The only exception is when the block fits on a single line:

```
if (true) do_something();
```

Reply:
We try to follow [style(9)](https://man.openbsd.org/style): "Use a space after keywords (if, while, for, return, switch). No braces are used for control statements with zero or only a single statement unless that statement is more than a single line, in which case they are permitted". Typically OpenBSD source has braces around all multiline statements, and omits them for simple one-liners.